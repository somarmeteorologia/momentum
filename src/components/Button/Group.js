import React, { Children, cloneElement, isValidElement } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { theme } from 'styled-tools'

const Container = styled.div`
  display: flex;

  .first {
    border-radius: ${theme('border.radius.four')} 0 0
      ${theme('border.radius.four')};
    border-right-width: 0;
    color: ${theme('colors.white.hundred')};
    background: ${theme('button.bg.group')};

    &:hover {
      border-right-width: 0;
    }
  }

  .middle {
    border-radius: 0;
    border-right-width: 0;
    border-color: ${theme('button.border.group')};

    &:hover {
      border-right-width: 0;
      border-color: ${theme('button.hover.primary.bg')};
      color: ${theme('colors.white.hundred')};
      background: ${theme('button.hover.primary.bg')};
    }
  }

  .last {
    border-color: ${theme('button.border.group')};
    border-radius: 0 ${theme('border.radius.four')}
      ${theme('border.radius.four')} 0;

    &:hover {
      color: ${theme('colors.white.hundred')};
      border-color: ${theme('button.hover.primary.bg')};
      background: ${theme('button.hover.primary.bg')};
    }
  }
`

const isValidButton = child =>
  isValidElement(child) && child.type.displayName === 'Button'

export default function Group({ children, orientation }) {
  const [first, ...childrens] = Children.map(children, child => {
    if (!isValidButton(child)) {
      throw new Error(
        'TypeError: Button Group component children is invalid, please use Button.'
      )
    }

    return child
  })

  const firstChildren = cloneElement(first, { className: 'first' })
  const last = cloneElement(childrens.pop(), {
    className: 'last',
    appearence: 'stroke'
  })

  const middle = childrens.map(child =>
    cloneElement(child, { className: 'middle', appearence: 'stroke' })
  )

  const identifiedChildrens = [firstChildren, ...middle, last]

  return <Container orientation={orientation}>{identifiedChildrens}</Container>
}

Group.defaultProps = {
  orientation: 'horizontal'
}

Group.propTypes = {
  orientation: PropTypes.oneOf(['vertical', 'horizontal'])
}
