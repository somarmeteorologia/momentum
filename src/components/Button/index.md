# Button

## About

## How use

```js
<Button appearence={Button.appearence.primary}>Primary</Button>
<Button appearence={Button.appearence.stroke}>Stroke</Button>
<Button appearence={Button.appearence.flat}>Flat</Button>
```
