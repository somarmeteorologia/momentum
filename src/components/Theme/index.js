import { tokens } from '@components/Tokens'

const { font, colors, border, zindex, animations, shadow } = tokens
const { white, ebony, cornflower, red, green, yellow, gray, ciano } = colors

const withCommons = theme => ({
  colors,
  font,
  border,
  zindex,
  animations,
  shadow,
  ...theme
})

export const Theme = {
  dark: withCommons({
    text: {
      primary: white.hundred,
      secondary: ebony.eighty,
      tertiary: cornflower.fifty,
      quaternary: ebony.seventy,
      disabled: ebony.sixty
    },
    bg: {
      primary: ebony.zero,
      secondary: ebony.thirteen,
      tertiary: ebony.sixteen
    },
    icons: {
      primary: white.hundred,
      secondary: cornflower.fifty
    },
    button: {
      bg: {
        primary: ebony.twenty,
        stroke: 'transparent',
        flat: 'transparent',
        rounded: white.hundred,
        disabled: ebony.twenty,
        group: 'transparent'
      },
      text: {
        primary: white.hundred,
        stroke: ebony.sixty,
        flat: white.hundred,
        group: ebony.sixty,
        rounded: cornflower.fifty,
        disabled: ebony.sixty
      },
      border: {
        primary: 'none',
        stroke: `1px solid ${ebony.sixty}`,
        flat: 'none',
        rounded: 'none',
        disabled: 'none',
        group: `1px solid ${ebony.twenty}`
      },
      hover: {
        primary: { bg: cornflower.fifty, text: white.hundred },
        stroke: { border: `1px solid ${white.hundred}`, text: white.hundred },
        flat: { bg: cornflower.fifty, text: white.hundred },
        group: { bg: ebony.twenty, text: white.hundred },
        rounded: { bg: white.hundred, text: cornflower.fifty }
      },
      active: {
        primary: cornflower.thirty,
        stroke: ebony.thirteen,
        flat: cornflower.thirty,
        group: { bg: ebony.twenty, border: 'none' }
      }
    },
    slider: {
      bg: {
        on: cornflower.fifty,
        off: ebony.twenty,
        thumb: white.hundred
      },
      text: {
        primary: white.hundred,
        secondary: white.hundred
      },
      border: {
        thumb: `2px solid ${ebony.seventy}`
      }
    },
    drawer: {
      bg: {
        primary: ebony.twenty
      },
      text: {
        primary: white.hundred,
        secondary: ebony.sixty
      },
      border: {
        primary: `1px solid ${ebony.sixteen}`
      },
      icon: {
        primary: white.hundred,
        secondary: cornflower.fifty
      },
      shadow: {
        primary: 'none'
      }
    },
    accordion: {
      bg: {
        primary: ebony.twenty
      },
      text: {
        primary: white.hundred
      },
      icon: {
        primary: ebony.sixty,
        secondary: white.hundred
      }
    },
    tabs: {
      text: {
        on: white.hundred,
        off: ebony.seventy
      },
      bg: {
        primary: 'transparent',
        secondary: cornflower.fifty
      }
    },
    selector: {
      text: {
        primary: white.hundred,
        disabled: ebony.sixty
      },
      border: {
        on: `1px solid ${cornflower.fifty}`,
        off: `1px solid ${ebony.sixty}`,
        disabled: `1px solid ${ebony.forty}`
      },
      bg: {
        on: cornflower.fifty,
        off: ebony.twenty
      },
      icon: {
        primary: white.hundred,
        disabled: ebony.forty
      }
    },
    tooltip: {
      text: {
        primary: ebony.zero,
        secondary: white.hundred
      },
      bg: {
        primary: white.hundred,
        secondary: cornflower.fifty
      },
      header: {
        primary: ebony.ninetySix,
        secondary: ebony.twenty
      },
      shadow: {
        primary: 'none'
      }
    },
    snackbar: {
      bg: {
        primary: ebony.zero
      },
      text: {
        primary: white.hundred,
        secondary: ciano.fifty
      }
    },
    notification: {
      bg: {
        primary: white.hundred
      },
      text: {
        primary: ebony.zero
      },
      icon: {
        primary: white.hundred,
        secondary: ebony.zero
      },
      close: {
        primary: ebony.zero
      },
      state: {
        primary: cornflower.fifty,
        danger: red.fifty,
        warning: yellow.fifty,
        success: green.fifty
      }
    },
    field: {
      bg: {
        primary: ebony.thirteen,
        secondary: white.hundred,
        tertiary: ebony.twenty,
        active: ebony.forty,
        hover: cornflower.fifty
      },
      text: {
        primary: white.hundred,
        secondary: ebony.sixty,
        tertiary: ebony.twenty,
        quaternary: white.hundred,
        active: cornflower.fifty,
        selected: white.hundred,
        danger: red.fifty,
        disabled: ebony.forty
      },
      border: {
        primary: `1px solid ${ebony.twenty}`,
        secondary: `1px solid ${ebony.forty}`,
        tertiary: '1px solid transparent',
        danger: `1px solid ${red.fifty}`
      },
      shadow: {
        primary: '0 2px 4px rgba(0, 0, 0, 0.1)'
      },
      icon: {
        primary: ebony.sixty,
        secondary: ebony.forty,
        tertiary: ebony.sixty,
        active: white.hundred,
        success: green.fifty
      },
      arrow: {
        primary: ebony.sixty,
        selected: white.hundred
      }
    },
    datepicker: {
      text: {
        primary: ebony.zero,
        secondary: white.hundred,
        tertiary: ebony.sixty,
        off: ebony.seventy
      },
      bg: {
        primary: white.hundred,
        secondary: cornflower.fifty,
        tertiary: ebony.thirteen
      },
      icon: {
        primary: cornflower.fifty,
        secondary: white.hundred,
        tertiary: ebony.sixty
      },
      shadow: {
        primary: 'none'
      },
      border: {
        primary: `1px solid ${ebony.twenty}`
      }
    },
    navigation: {
      bg: {
        primary: ebony.thirteen,
        secondary: ebony.sixteen,
        tertiary: ebony.sixty,
        hover: white.hundred
      },
      separator: {
        primary: ebony.sixteen
      },
      text: {
        primary: white.hundred,
        hover: cornflower.fifty
      },
      shadow: {
        primary: 'none'
      }
    },
    menu: {
      bg: {
        primary: ebony.thirteen
      },
      toggle: {
        primary: ebony.sixteen,
        secondary: ebony.thirteen,
        tertiary: white.hundred
      }
    },
    player: {
      text: {
        primary: white.hundred,
        secondary: ebony.zero
      },
      bg: {
        primary: ebony.twenty,
        secondary: white.hundred,
        opacity: '0.8'
      },
      icon: {
        primary: white.hundred,
        secondary: ebony.twenty
      },
      progress: {
        on: white.hundred,
        off: gray.fifty
      },
      shadow: {
        primary: 'none'
      }
    },
    datatable: {
      text: {
        primary: white.hundred,
        hover: ebony.zero
      },
      bg: {
        primary: ebony.twenty,
        rowHover: white.hundred,
        columnHover: ebony.sixteen
      },
      border: {
        primary: `1px solid ${ebony.thirteen}`
      },
      icon: {
        primary: ebony.sixty
      },
      pagination: {
        bg: {
          primary: ebony.zero,
          secondary: ebony.twenty,
          tertiary: white.hundred,
          active: cornflower.fifty
        },
        text: {
          primary: ebony.seventy,
          secondary: white.hundred,
          active: white.hundred
        },
        border: {
          primary: 'none'
        }
      }
    }
  }),
  light: withCommons({
    text: {
      primary: ebony.zero,
      secondary: ebony.twenty,
      tertiary: cornflower.fifty,
      quaternary: ebony.seventy,
      disabled: ebony.sixty
    },
    bg: {
      primary: white.hundred,
      secondary: ebony.ninetySix,
      tertiary: ebony.ninety
    },
    icons: {
      primary: ebony.zero,
      secondary: cornflower.fifty
    },
    button: {
      bg: {
        primary: ebony.ninetySix,
        stroke: 'transparent',
        flat: 'transparent',
        rounded: white.hundred,
        disabled: ebony.ninetySix,
        group: 'transparent'
      },
      text: {
        primary: ebony.zero,
        stroke: ebony.sixty,
        flat: ebony.zero,
        group: ebony.eighty,
        rounded: cornflower.fifty,
        disabled: ebony.sixty
      },
      border: {
        primary: 'none',
        stroke: `1px solid ${ebony.sixty}`,
        flat: 'none',
        rounded: 'none',
        disabled: 'none',
        group: `1px solid ${ebony.eighty}`
      },
      hover: {
        primary: { bg: cornflower.fifty, text: white.hundred },
        stroke: {
          border: `1px solid ${cornflower.fifty}`,
          text: cornflower.fifty
        },
        flat: { bg: cornflower.fifty, text: white.hundred },
        group: { bg: ebony.sixty, text: white.hundred },
        rounded: { bg: white.hundred, text: cornflower.fifty }
      },
      active: {
        primary: cornflower.thirty,
        stroke: ebony.ninetySix,
        flat: cornflower.thirty,
        group: { bg: ebony.sixty, border: 'none' }
      }
    },
    slider: {
      bg: {
        on: cornflower.fifty,
        off: ebony.ninety,
        thumb: white.hundred
      },
      text: {
        primary: ebony.zero,
        secondary: cornflower.fifty
      },
      border: {
        thumb: `2px solid ${ebony.eighty}`
      }
    },
    drawer: {
      bg: {
        primary: white.hundred
      },
      text: {
        primary: ebony.zero,
        secondary: ebony.sixty
      },
      border: {
        primary: `1px solid ${ebony.ninety}`
      },
      icon: {
        primary: cornflower.fifty,
        secondary: ebony.zero
      },
      shadow: {
        primary: `0 4px 4px 0 ${shadow.default}`
      }
    },
    accordion: {
      bg: {
        primary: ebony.ninetySix
      },
      text: {
        primary: ebony.zero
      },
      icon: {
        primary: ebony.sixty,
        secondary: ebony.zero
      }
    },
    tabs: {
      text: {
        on: white.hundred,
        off: ebony.seventy
      },
      bg: {
        primary: 'transparent',
        secondary: cornflower.fifty
      }
    },
    selector: {
      text: {
        primary: ebony.zero,
        disabled: ebony.sixty
      },
      border: {
        on: `1px solid ${cornflower.fifty}`,
        off: `1px solid ${ebony.sixty}`,
        disabled: `1px solid ${ebony.eighty}`
      },
      bg: {
        on: cornflower.fifty,
        off: ebony.ninety
      },
      icon: {
        primary: white.hundred,
        disabled: ebony.eighty
      }
    },
    tooltip: {
      text: {
        primary: cornflower.fifty,
        secondary: white.hundred
      },
      bg: {
        primary: white.hundred,
        secondary: cornflower.fifty
      },
      header: {
        primary: ebony.ninetySix,
        secondary: ebony.twenty
      },
      shadow: {
        primary: `0 2px 4px 0 ${shadow.default}`
      }
    },
    snackbar: {
      bg: {
        primary: ebony.zero
      },
      text: {
        primary: white.hundred,
        secondary: ciano.fifty
      }
    },
    notification: {
      bg: {
        primary: ebony.thirteen
      },
      text: {
        primary: white.hundred
      },
      icon: {
        primary: white.hundred,
        secondary: ebony.zero
      },
      close: {
        primary: white.hundred
      },
      state: {
        primary: cornflower.fifty,
        danger: red.fifty,
        warning: yellow.fifty,
        success: green.fifty
      }
    },
    field: {
      bg: {
        primary: ebony.ninetySix,
        secondary: white.hundred,
        tertiary: ebony.eighty,
        active: cornflower.fifty,
        hover: cornflower.fifty
      },
      text: {
        primary: ebony.twenty,
        secondary: ebony.sixty,
        tertiary: ebony.twenty,
        quaternary: white.hundred,
        active: cornflower.fifty,
        selected: ebony.twenty,
        disabled: ebony.eighty,
        danger: red.fifty
      },
      border: {
        primary: `1px solid ${ebony.eighty}`,
        secondary: `1px solid ${ebony.sixty}`,
        tertiary: `1px solid ${ebony.ninetySix}`,
        danger: `1px solid ${red.fifty}`
      },
      shadow: {
        primary: `0 2px 4px ${shadow.default}`
      },
      arrow: {
        primary: ebony.sixty,
        selected: ebony.twenty
      },
      icon: {
        primary: ebony.sixty,
        secondary: white.hundred,
        tertiary: ebony.sixty,
        active: white.hundred,
        success: green.fifty
      }
    },
    datepicker: {
      text: {
        primary: ebony.zero,
        secondary: white.hundred,
        tertiary: ebony.sixty,
        off: ebony.seventy
      },
      bg: {
        primary: white.hundred,
        secondary: cornflower.fifty,
        tertiary: ebony.ninetySix
      },
      icon: {
        primary: cornflower.fifty,
        secondary: white.hundred,
        tertiary: ebony.sixty
      },
      shadow: {
        primary: `0 2px 4px ${shadow.default}`
      },
      border: {
        primary: `1px solid ${ebony.ninety}`
      }
    },
    navigation: {
      bg: {
        primary: ebony.ninetySix,
        secondary: ebony.ninety,
        tertiary: ebony.eighty,
        hover: white.hundred
      },
      separator: {
        primary: ebony.ninety
      },
      text: {
        primary: ebony.twenty,
        hover: cornflower.fifty
      },
      shadow: {
        primary: `0 0 16px ${shadow.darker}`
      }
    },
    menu: {
      bg: {
        primary: ebony.ninetySix
      },
      toggle: {
        primary: white.hundred,
        secondary: ebony.ninety,
        tertiary: cornflower.fifty
      }
    },
    player: {
      text: {
        primary: white.hundred,
        secondary: ebony.zero
      },
      bg: {
        primary: ebony.zero,
        secondary: white.hundred,
        opacity: '0.8'
      },
      icon: {
        primary: white.hundred,
        secondary: ebony.zero
      },
      progress: {
        on: white.hundred,
        off: gray.fifty
      },
      shadow: {
        primary: `0px 0px 4px 0px ${shadow.default}`
      }
    },
    datatable: {
      text: {
        primary: ebony.zero,
        hover: ebony.zero
      },
      bg: {
        primary: white.hundred,
        rowHover: ebony.ninetySix,
        columnHover: ebony.ninetySix
      },
      border: {
        primary: `1px solid ${ebony.ninety}`
      },
      icon: {
        primary: ebony.sixty
      },
      pagination: {
        bg: {
          primary: white.hundred,
          secondary: ebony.ninetySix,
          tertiary: white.hundred,
          active: cornflower.fifty
        },
        text: {
          primary: ebony.seventy,
          secondary: cornflower.fifty,
          active: white.hundred
        },
        border: {
          primary: `1px solid ${ebony.ninety}`
        }
      }
    }
  })
}
