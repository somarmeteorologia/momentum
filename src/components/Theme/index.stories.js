// import React from 'react'
// import { storiesOf } from '@storybook/react'

// import { GROUPS } from '@helpers/hierarchySeparators'

// import { Reset } from '@components/Reset'

// // storiesOf(`${GROUPS.TOKENS}|Color`, module).addDecorator(story => (
// //   <>
// //     <Reset />
// //     {story()}
// //   </>
// // ))
// // .add('Default', () => <Colors colors={colors} />)
