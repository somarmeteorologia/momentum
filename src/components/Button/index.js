import React, { memo, useEffect, useState, useContext } from 'react'
import PropTypes from 'prop-types'
import { switchProp, ifProp, prop } from 'styled-tools'
import styled, { css, ThemeContext } from 'styled-components'

import { Icon as Iconable } from '@components/Icon'

import Group from './Group'

const Icon = styled.div`
  margin-right: 12px;
  display: flex;
  align-items: center;
`

const Content = styled.button`
  border-radius: ${prop('theme.border.radius.four')};
  font-family: ${prop('theme.font.family.inter')};
  font-weight: ${prop('theme.font.weight.bold')};
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0 16px;
  cursor: pointer;
  width: ${ifProp('full', '100%', 'unset')};
  text-transform: ${ifProp('uppercase', 'uppercase', 'none')};

  ${switchProp('size', {
    small: css`
      height: 32px;
      font-size: ${prop('theme.font.size.twelve')};
    `,
    default: css`
      height: 40px;
      font-size: ${prop('theme.font.size.fourteen')};
    `,
    large: css`
      height: 48px;
      font-size: ${prop('theme.font.size.sixteen')};
    `
  })};

  ${switchProp('appearence', {
    primary: css`
      background-color: ${prop('theme.button.bg.primary')};
      color: ${prop('theme.button.text.primary')};
      border: ${prop('theme.button.border.primary')};

      &:hover {
        background-color: ${prop('theme.button.hover.primary.bg')};
        color: ${prop('theme.button.hover.primary.text')};
      }

      &:active {
        background-color: ${prop('theme.button.active.primary')};
      }
    `,
    stroke: css`
      background-color: ${prop('theme.button.bg.stroke')};
      color: ${prop('theme.button.text.stroke')};
      border: ${prop('theme.button.border.stroke')};

      &:hover {
        color: ${prop('theme.button.hover.stroke.text')};
        border: ${prop('theme.button.hover.stroke.border')};
      }

      &:active {
        background-color: ${prop('theme.button.active.stroke')};
      }
    `,
    flat: css`
      background-color: ${prop('theme.button.bg.flat')};
      color: ${prop('theme.button.text.flat')};
      border: ${prop('theme.button.border.flat')};

      &:hover {
        background-color: ${prop('theme.button.hover.flat.bg')};
        color: ${prop('theme.button.hover.flat.text')};
      }

      &:active {
        background-color: ${prop('theme.button.active.flat')};
      }
    `
  })};

  ${ifProp(
    { disabled: true },
    css`
      background-color: ${prop('theme.button.bg.disabled')};
      color: ${prop('theme.button.text.disabled')};
      border: ${prop('theme.button.border.disabled')};

      &:hover {
        background-color: ${prop('theme.button.bg.disabled')};
        color: ${prop('theme.button.text.disabled')};
        border: ${prop('theme.button.border.disabled')};
      }
    `
  )}

  &:focus {
    outline: none;
  }

  &:hover {
    transition: all linear 0.2s;
  }
`

const types = {
  primary: 'primary',
  stroke: 'stroke',
  flat: 'flat'
}

export const Button = memo(
  ({
    appearence,
    size,
    disabled,
    full,
    uppercase,
    loading,
    icon,
    children,
    onClick,
    ...props
  }) => {
    const { button } = useContext(ThemeContext)
    const [color, setColor] = useState('')

    const setHoverColor = () =>
      types[appearence] && setColor(button.hover[appearence].text)

    const setInitialColor = () =>
      types[appearence] && setColor(button.text[appearence])

    useEffect(() => {
      setInitialColor()
    }, [button])

    return (
    <Content
        appearence={appearence}
        size={size}
        disabled={disabled}
        full={full}
        uppercase={uppercase}
        onClick={onClick}
        onMouseEnter={setHoverColor}
        onMouseLeave={setInitialColor}
        {...props}
      >
        {loading ? (
          <>
            <Icon>
              <Iconable name={Iconable.name.loading} color={color} />
            </Icon>
            Loading
          </>
        ) : (
          <>
            {icon && <Icon>{icon(color)}</Icon>}
            {children}
          </>
        )}
      </Content>
    )
  }
)

Button.Group = Group

Button.displayName = 'Button'

Button.defaultProps = {
  appearence: 'primary',
  size: 'default',
  disabled: false,
  full: false,
  uppercase: false,
  loading: false,
  onClick: () => {}
}

Button.propTypes = {
  appearence: PropTypes.oneOf(['primary', 'stroke', 'flat']),
  size: PropTypes.oneOf(['small', 'default', 'large']),
  disabled: PropTypes.bool,
  icon: PropTypes.func,
  onClick: PropTypes.func,
  full: PropTypes.bool,
  uppercase: PropTypes.bool,
  loading: PropTypes.bool
}

Button.appearence = {
  primary: 'primary',
  stroke: 'stroke',
  flat: 'flat'
}

Button.size = {
  small: 'small',
  default: 'default',
  large: 'large'
}
