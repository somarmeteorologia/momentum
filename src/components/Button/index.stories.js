import React from 'react'
import { storiesOf } from '@storybook/react'

import { GROUPS } from '@helpers/hierarchySeparators'
import { Container } from '@helpers/components/Container'

import { Reset } from '@components/Reset'
import { Button } from '@components/Button'
import { Icon } from '@components/Icon'

import markdown from './index.md'

storiesOf(`${GROUPS.COMPONENTS}|Button`, module)
  .addDecorator(story => (
    <Container>
      <Reset />
      {story()}
    </Container>
  ))
  .add(
    'Appearence',
    () => (
      <>
        <Button appearence={Button.appearence.primary}>Primary</Button>
        <Button appearence={Button.appearence.stroke}>Stroke</Button>
        <Button appearence={Button.appearence.flat}>Flat</Button>
        <Button appearence={Button.appearence.rounded}>Rounded</Button>
      </>
    ),
    {
      notes: {
        markdown
      }
    }
  )
  .add('Disabled', () => (
    <>
      <Button disabled={true}>Disabled</Button>
    </>
  ))
  .add('With link', () => (
    <>
      <Button href="https://google.com">Open Google</Button>
    </>
  ))
  .add('Size', () => (
    <>
      <Button size={Button.size.small}>Small</Button>
      <Button size={Button.size.default}>Default</Button>
      <Button size={Button.size.large}>Large</Button>
    </>
  ))
  .add('Full', () => (
    <>
      <Button full={true}>Full</Button>
    </>
  ))
  .add('With icon', () => (
    <>
      <Button
        icon={color => (
          <Icon
            name={Icon.name.download}
            width={22}
            height={22}
            color={color}
          />
        )}
      >
        Primary
      </Button>
      <Button
        appearence={Button.appearence.stroke}
        icon={color => (
          <Icon
            name={Icon.name.download}
            width={22}
            height={22}
            color={color}
          />
        )}
      >
        Stroke
      </Button>
      <Button
        appearence={Button.appearence.flat}
        icon={color => (
          <Icon
            name={Icon.name.download}
            width={22}
            height={22}
            color={color}
          />
        )}
      >
        Flat
      </Button>
    </>
  ))
  .add('Loading', () => (
    <>
      <Button loading={true} />
    </>
  ))
  .add('Group', () => (
    <>
      <Button.Group activeId="Primary">
        <Button id="Primary">Primary</Button>
        <Button id="Secondary">Secondary</Button>
        <Button id="Tertiary">Tertiary</Button>
      </Button.Group>
      <Button.Group
        activeId="Primary"
        appearence={Button.Group.appearence.secondary}
      >
        <Button id="Primary">Primary</Button>
        <Button id="Secondary">Secondary</Button>
        <Button id="Tertiary">Tertiary</Button>
      </Button.Group>
    </>
  ))
  .add('Group vertical', () => (
    <>
      <Button.Group
        activeId="Primary"
        orientation={Button.Group.orientation.vertical}
      >
        <Button id="Primary">Primary</Button>
        <Button id="Secondary">Secondary</Button>
        <Button id="Tertiary">Tertiary</Button>
      </Button.Group>
      <Button.Group
        activeId="Primary"
        orientation={Button.Group.orientation.vertical}
        appearence={Button.Group.appearence.secondary}
      >
        <Button id="Primary">Primary</Button>
        <Button id="Secondary">Secondary</Button>
        <Button id="Tertiary">Tertiary</Button>
      </Button.Group>
    </>
  ))
