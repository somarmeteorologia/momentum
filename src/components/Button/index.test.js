import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import { Theme } from '@components/Theme'
import { Button } from '@components/Button'

const { button, font } = Theme.light

const withProvider = ({ ...props }) => (
  <ThemeProvider theme={Theme.light}>
    <Button {...props}>Primary</Button>
  </ThemeProvider>
)

test('Not expect visual regression', () => {
  const { container } = render(withProvider({}))

  expect(container.firstChild).toMatchSnapshot()
})

test('Expect correct appearances', () => {
  const { container, rerender } = render(withProvider({}))

  expect(container.firstChild).toHaveStyleRule(
    'background-color',
    button.bg.primary
  )

  rerender(withProvider({ appearence: Button.appearence.stroke }))

  expect(container.firstChild).toHaveStyleRule(
    'background-color',
    button.bg.stroke
  )
})

test('Expect full width when with prop', () => {
  const { container, rerender } = render(withProvider())

  expect(container.firstChild).toHaveStyleRule('width', 'unset')

  rerender(withProvider({ full: true }))

  expect(container.firstChild).toHaveStyleRule('width', '100%')
})

test('Expect correct sizes', () => {
  const { container, rerender } = render(withProvider({}))

  expect(container.firstChild).toHaveStyleRule('height', '40px')
  expect(container.firstChild).toHaveStyleRule('font-size', font.size.fourteen)

  rerender(withProvider({ size: Button.size.small }))

  expect(container.firstChild).toHaveStyleRule('height', '32px')
  expect(container.firstChild).toHaveStyleRule('font-size', font.size.twelve)

  rerender(withProvider({ size: Button.size.large }))

  expect(container.firstChild).toHaveStyleRule('height', '48px')
  expect(container.firstChild).toHaveStyleRule('font-size', font.size.sixteen)
})
