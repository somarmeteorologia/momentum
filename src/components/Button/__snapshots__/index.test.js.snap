// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not expect visual regression 1`] = `
.c0 {
  border-radius: 0.25rem;
  font-family: Inter,sans-serif;
  font-weight: 600;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 16px;
  cursor: pointer;
  width: unset;
  text-transform: none;
  height: 40px;
  font-size: 0.875rem;
  background-color: #f1f3fA;
  color: #080B19;
  border: none;
}

.c0:hover {
  background-color: #133BEC;
  color: #fff;
}

.c0:active {
  background-color: #0B238E;
}

.c0:focus {
  outline: none;
}

.c0:hover {
  -webkit-transition: all linear 0.2s;
  transition: all linear 0.2s;
}

<button
  class="c0"
>
  Primary
</button>
`;
